sklearn/utils/validation.py

```diff
@@ -336,7 +337,8 @@ def check_array(array, accept_sparse=None, dtype=None, order=None, copy=False,
                              " minimum of %d is required."
                              % (n_samples, shape_repr, ensure_min_samples))
 
-    if ensure_min_features > 0 and ensure_2d and not allow_nd:
+
+    if ensure_min_features > 0 and array.ndim == 2:
```

Comment:
There is a bunch of common checks that would fail otherwise. I can add a couple of unittest in `test_validation.py` to make it more explicit.